packages/typescript/src/core/api.ts

```diff
@@ -0,0 +1,106 @@
+import { trace, context, Span as ApiSpan } from '@opentelemetry/api';
+import { Span as OTelSpan } from '@opentelemetry/sdk-trace-node';
+import { SpanType } from './constants';
+import { createMlflowSpan, LiveSpan, NoOpSpan } from './entities/span';
+import { getTracer } from './provider';
+import { InMemoryTraceManager } from './trace_manager';
+import { convertNanoSecondsToHrTime } from './utils';
+
+/*
+ * Options for starting a span
+ *
+ * @param name The name of the span.
+ * @param span_type The type of the span.
+ * @param inputs The inputs of the span.
+ * @param attributes The attributes of the span.
+ * @param startTimeNs The start time of the span in nanoseconds. If not provided, the current time will be used.
+ * @param parent The parent span object.
+ */
+export interface SpanOptions {
+  name: string;
+  span_type?: SpanType;
+  inputs?: any;
+  attributes?: Record<string, any>;
+  startTimeNs?: number;
+  parent?: LiveSpan;
+}
+
+/**
+ * Start a new span with the given name and span type.
+ *
+ * This function does NOT attach the created span to the current context.
+ * The span must be ended by calling `end` method on the returned Span object.
+ */
+export function startSpan(options: SpanOptions): LiveSpan {
```

Comment:
Let's use destructuring
```suggestion
export function startSpan({
   name,
   span_type,
   inputs,
   attributes,
   startTimeNs,
   parent,
}: SpanOptions): LiveSpan {
```

Reply:
This option will be shared with `withSpan` API added in the next PR🙂